### PR TITLE
Softmax bug fix

### PR DESCRIPTION
**Pull request by erstrong**
Correction for how the softmax algorithm normalizes the values used in exponent calculations. This change impacts unit tests, doc tests, and example code.

Also removed the matplotlib backend specification in the simulator as this is not compatible with latest Mac OS versions, and some example scripts hadn't been updated for parallelization backend.

### CODE REVIEW

File: docs/about.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
   
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   
-  <title>About Multi-Armed Bandits &mdash; MABWiser 1.0.0 documentation</title>
+  <title>About Multi-Armed Bandits &mdash; MABWiser 1.8.1 documentation</title>
```
Comment (skadio) [inline docs/about.html:11]: ah thank you!

Comment (erstrong) [reply docs/about.html:11]: Can't take credit, @dorukkilitcioglu previously pointed it out :)

File: mabwiser/softmax.py [hunk 2]
```diff
@@ -63,14 +63,11 @@ def _expectation_operation(self):
...
-        min_mean = min(self.arm_to_mean.values())
-        scaling_range = max_mean - min_mean
-        if scaling_range == 0:
-            scaling_range = 1
 
         # Scale the means and calculate the natural exponents --decrement max to avoid overflow from np.exp(x)
+        # Reference: https://stackoverflow.com/questions/42599498/numercially-stable-softmax
         for arm in self.arm_to_exponent:
-            self.arm_to_exponent[arm] = math.exp((self.arm_to_mean[arm] - max_mean) / (scaling_range * self.tau))
+            self.arm_to_exponent[arm] = math.exp((self.arm_to_mean[arm] - max_mean) / self.tau)
```
Comment (skadio) [inline mabwiser/softmax.py:70]: cool.

From that reference, it seems SciPy also offers softmax. https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.softmax.html#scipy.special.softmax 

It doesn't seem to have a "tau" parameter. 

Are we changing the formula in the docs as well?
https://fmr-llc.github.io/mabwiser/api.html#mabwiser.mab.LearningPolicy.Softmax

Comment (erstrong) [reply mabwiser/softmax.py:70]: Tau is from the Kuleshov multi-armed bandits paper. The formula is updated in the docs, it won't show on the website version until this pr is merged.

Comment (erstrong) [reply mabwiser/softmax.py:70]: This is the paper: https://arxiv.org/abs/1402.6028. It explains tau is used to control how random the choice is. This is the reinforcement learning variation on a softmax function.

**Pull request merged**